server/utils/smartSearch.js

```diff
@@ -0,0 +1,42 @@
+//this function compares the list of user following and followers (if they exist)
+//and returns the users that follow the user but the user does not follow back and
+//a list of users that should be excluded from our broader search so we dont't get
+//any repititive data
+
+const smartSearch = (userRelations, loggedInUserId) => {
+	//this object is created to make the code more efficient in terms of time complexity
+	let userFollowingList = {};
+
+	//this is a list of people that follow the user but the user does not follow back
+	let noFollowBack = [];
+
+	//This is a list of ids that should be excluded when we want to search all the users
+	let excludeFromSearch = [loggedInUserId];
```

Comment:
These can all be const and this should all still work

Reply:
changed to const